### PR TITLE
Partially addresses 'alot refreshes search list during execution of command list' #769 

### PR DESCRIPTION
**Pull request by naegling**
Both the OP and my own issue involved atomically adding and removing tags that could effect the contents of search.  This patch adds a new command `applytags` with arguments formatted as in `notmuch tag` command.

The OP's example:
```
[bindings]
    [[search]]
        delete = "tag deleted; untag undeleted,unread,inbox;"
        insert = "tag undeleted; untag deleted;"
```
could be configured as:
```
[bindings]
    [[search]]
        delete = "applytags +deleted -undeleted -unread -inbox;"
        insert = "applytags +undeleted -deleted;"
```

### CODE REVIEW

**Comment by naegling** [comment]
Well, that's annoying.  I just noticed that the first argument to applytags has to be an added tag.  Otherwise, its interpreted as a (invalid) command line option.

**Comment by pazz** [comment]
hehe yes, lols. never thought of that :D

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (kbingham) [inline .vscode/launch.json]: I don't think this file should be added.

**Comment by kbingham** [comment]
This is a really nice idea, I've also hit the tags affecting the wrong thread after a refresh.

File: .vscode/launch.json (hunk 1 above)
Comment (naegling) [reply .vscode/launch.json]: I've removed the .vscode file accidentally added.

**Comment by naegling** [comment]
> Well, that's annoying. I just noticed that the first argument to applytags has to be an added tag. Otherwise, its interpreted as a (invalid) command line option.

Suppressing the option interpretation looks both invasive and hacky.  Currently, I see 2 options for more consistent syntax. to remove the dick tag and add the jane tag:
1. use a standalone + as a noop.  `applytags + -dick +jane`.
2. introduce an option name.  `applytags -ops -dick +jane`. The option name could be optional, so that `applytags +jane -dick` would still work as before.

Comments Welcomed!

**Pull request closed without merging**